Update template to Webpack 5 for better experience

non25 opened the pull request:
I find it sad that there's 18k weekly downloads of `svelte-loader` and these users don't get the best possible svelte experience on Webpack. This template doesn't show a good example of how this could be done either, and it is the most important one, because it is official.

This fixes the situation by updating `svelte-loader` from master, which has Webpack 5 and HMR support and also updating everything else to the most recent version. I believe starters don't want to use stale deps.

Issues: Fixes #12, Fixes #31, Fixes #34
PRs: Fixes #33, Fixes #24, Fixes #21, Fixes #28

non25 committed:
Update template to Webpack 5 for better experience

benmccann commented:
looks like a good idea. we will need to wait for a new release of `svelte-loader`

benmccann commented on package.json:
```diff
@@ -2,19 +2,18 @@
+    "mini-css-extract-plugin": "^1.3.4",
+    "style-loader": "^2.0.0",
+    "svelte": "^3.31.2",
+    "svelte-loader": "sveltejs/svelte-loader",
```
we should make this `^3.0.0`

benmccann commented:
The new version of `svelte-loader` has been released now

non25 committed:
Use released svelte-loader 3.0.0 from npm

benmccann commented on webpack.config.js:
```diff
@@ -51,5 +61,8 @@ module.exports = {
 		})
 	],
-	devtool: prod ? false: 'source-map'
+	devtool: prod ? false: 'source-map',
```
```suggestion
	devtool: prod ? false : 'source-map',
```

non25 committed:
false: -> false :

Co-authored-by: Ben McCann <322311+benmccann@users.noreply.github.com>

non25 committed:
Move global.css to src, always use mini-css-extract-plugin

Make HMR work for global.css
Use mini-css-extract-plugin instead of style-loader, it supports HMR
Get rid of the need to have bundle.css by prebuilding, making it
empty or what else

non25 commented:
I've been thinking about `emitCss: false` for HMR to work and somehow having `bundle.css` not to break reloading. The ways to solve this are:

1. Prebuild `bundle.css` before running devmode. Drawbacks: old css from previous production build interfering with new, need to prebuild stuff
2. Make an empty `bundle.css` in public before running devmode. Drawbacks: kinda manualish or could introduce wierd stuff in `webpack.config.js` to make it automatic
3. Move `global.css` as an import to the `main.js`. Can't see drawbacks for that, allows HMR to work for `global.css`, fixes #31. 

Currently, components don't output css in devmode, and there's no global imported css, so by default there won't be any `bundle.css` in devmode in real/virtual fs, so hot reloading will break.

^ That is also because we are using `style-loader`. Moving `global.css` to src won't fix #12 by itself.
We will also need to use `mini-css-extract-plugin` in devmode, but that is preferable nowadays anyway.

non25 committed:
style-loader is not used anymore, might as well remove it

non25 commented:
Also I think #4, #9, #26, #30, #32 are irrelevant now and could be closed.

non25 committed:
Use same file structure as svelte rollup template

non25 committed:
Eslint no-path-concat fix

non25 committed:
Copy index.html, App.svelte, global.css from rollup template
